inc/assets.php

```diff
@@ -220,3 +221,33 @@ function enqueue_frontend_styles() : void {
 		wp_enqueue_style( 'mapbox-css', 'https://api.mapbox.com/mapbox-gl-js/v3.2.0/mapbox-gl.css' );
 	}
 }
+
+/**
+ * Intercept URIs for enqueued CSS and rewrite version strings for plugin assets
+ * to ensure a hash-based version string is in use.
+ *
+ * Removes requirement for constituent block versions to be incremented on each
+ * related code update, which is difficult to enforce in process.
+ *
+ * @param string $src The source URL of the enqueued style.
+ * @return string Filtered URL.
+ */
+function cachebust_stylesheets( string $src ): string {
+	$plugin_base_url = trailingslashit( plugin_dir_url( __DIR__ ) );
+	if ( strpos( $src, $plugin_base_url ) !== 0 ) {
+		return $src;
+	}
+
+	$file_relative_path = preg_replace( '/\?.*$/', '', str_replace( $plugin_base_url, '', $src ) );
+	$file_path          = plugin_dir_path( __DIR__ ) . $file_relative_path;
+	if ( ! is_readable( $file_path ) ) {
+		return $src;
+	}
+
+	$version = hash_file( 'crc32', $file_path ) ?: filemtime( $file_path );
```

Comment:
These could use some comments to make it more quickly obvious what you're checking for.